packages/core/src/browser/xhrProxy.spec.ts

```diff
@@ -110,9 +110,9 @@ describe('xhr proxy', () => {
     const onReadyStateChange = jasmine.createSpy()
     withXhr({
       setup(xhr) {
-        xhr.onreadystatechange = onReadyStateChange
-        xhr.addEventListener('load', () => xhr.abort())
         xhr.open('GET', '/ok')
+        xhr.addEventListener('readystatechange', onReadyStateChange)
+        xhr.addEventListener('load', () => xhr.abort())
         xhr.send()
```

Comment:
FMU, this test is related to the Angular use case. By looking at their implementation [here](https://github.com/angular/angular/blob/master/packages/common/http/src/xhr.ts).
I can't find any readystatechange handler in their code. What I found, is that they [abort in-flight requests in the unsubscribe on their observable](https://github.com/angular/angular/blob/3d1c905c9190f467cc6f8b0c3c612cdd203c3f75/packages/common/http/src/xhr.ts#L317) by looking at the xhr `readyState`. Both the previous implementation and the new one work because onreadystatechange is called before the xhr apply internally the DONE readyState.

That's why I changed the test to be closer to the Angular implementation.

Reply:
what about supporting both then?

Reply:
I'm working on it